kythe/cxx/common/kzip_writer.cc

```diff
@@ -52,8 +55,16 @@ Status WriteTextFile(zip_t* archive, const std::string& path,
                      absl::string_view content) {
   if (auto source =
           zip_source_buffer(archive, content.data(), content.size(), 0)) {
-    if (zip_file_add(archive, path.c_str(), source, ZIP_FL_ENC_UTF_8) >= 0) {
-      return OkStatus();
+    if (auto idx =
+            zip_file_add(archive, path.c_str(), source, ZIP_FL_ENC_UTF_8);
+        idx >= 0) {
+      if (idx == 0) {
```

Comment:
Why is zero being treated special? It shouldn't be different than any other index.

Reply:
I think 0 is only returned in the case of zip_file_replace, but it should be harmless to handle 0 the same as positive indexes since otherwise the case would do something unexpected silently.